src/core/tools/generateImageTool.ts

```diff
@@ -0,0 +1,264 @@
+import * as vscode from "vscode"
+import { Task } from "../task/Task"
+import { formatResponse } from "../prompts/responses"
+import { ToolUse, AskApproval, HandleError, PushToolResult, RemoveClosingTag } from "../../shared/tools"
+import { fileExistsAtPath } from "../../utils/fs"
+import { getReadablePath } from "../../utils/path"
+import { isPathOutsideWorkspace } from "../../utils/pathUtils"
+import { EXPERIMENT_IDS, experiments } from "../../shared/experiments"
+import { safeWriteJson } from "../../utils/safeWriteJson"
+
+// Hardcoded list of image generation models for now
+const IMAGE_GENERATION_MODELS = [
+	"google/gemini-2.5-flash-image-preview",
+	// Add more models as they become available
+]
+
+interface ImageGenerationResponse {
+	choices?: Array<{
+		message?: {
+			content?: string
+			images?: Array<{
+				type?: string
+				image_url?: {
+					url?: string
+				}
+			}>
+		}
+	}>
+	error?: {
+		message?: string
+		type?: string
+		code?: string
+	}
+}
+
+export async function generateImageTool(
+	cline: Task,
+	block: ToolUse,
+	askApproval: AskApproval,
+	handleError: HandleError,
+	pushToolResult: PushToolResult,
+	removeClosingTag: RemoveClosingTag,
+) {
+	const prompt: string | undefined = block.params.prompt
+	const relPath: string | undefined = block.params.path
+
+	// Check if the experiment is enabled
+	const provider = cline.providerRef.deref()
+	const state = await provider?.getState()
+	const isImageGenerationEnabled = experiments.isEnabled(state?.experiments ?? {}, EXPERIMENT_IDS.IMAGE_GENERATION)
+
+	if (!isImageGenerationEnabled) {
+		pushToolResult(
+			formatResponse.toolError(
+				"Image generation is an experimental feature that must be enabled in settings. Please enable 'Image Generation' in the Experimental Settings section.",
+			),
+		)
+		return
+	}
+
+	if (block.partial && (!prompt || !relPath)) {
+		// Wait for complete parameters
+		return
+	}
+
+	if (!prompt) {
+		cline.consecutiveMistakeCount++
+		cline.recordToolError("generate_image")
+		pushToolResult(await cline.sayAndCreateMissingParamError("generate_image", "prompt"))
+		return
+	}
+
+	if (!relPath) {
+		cline.consecutiveMistakeCount++
+		cline.recordToolError("generate_image")
+		pushToolResult(await cline.sayAndCreateMissingParamError("generate_image", "path"))
+		return
+	}
+
+	// Validate access permissions
+	const accessAllowed = cline.rooIgnoreController?.validateAccess(relPath)
+	if (!accessAllowed) {
+		await cline.say("rooignore_error", relPath)
+		pushToolResult(formatResponse.toolError(formatResponse.rooIgnoreError(relPath)))
+		return
+	}
+
+	// Check if file is write-protected
+	const isWriteProtected = cline.rooProtectedController?.isWriteProtected(relPath) || false
+
+	// Get OpenRouter API key from settings or profile
+	const imageGenerationSettings = (state as any)?.imageGenerationSettings
+	let openRouterApiKey = imageGenerationSettings?.openRouterApiKey
+
+	// If no API key in settings, check profiles for openRouterApiKey
```

Comment:
Not sure about this - we might want people to explicitly set a key with the experimental setting